Header header
string is_available
uint8[64] temp_map
